lib/streamlit/elements/alert.py

```diff
@@ -12,86 +12,147 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import cast, TYPE_CHECKING
+from typing import cast, Optional, TYPE_CHECKING
 
 from streamlit.proto.Alert_pb2 import Alert as AlertProto
+from streamlit.errors import StreamlitAPIException
+from streamlit.string_util import is_emoji_valid
 from .utils import clean_text
+import re
 
 if TYPE_CHECKING:
     from streamlit.delta_generator import DeltaGenerator
     from streamlit.type_util import SupportsStr
 
+# Function to check the icon parameter on the alert.
+# We check if what's been added is a valid emoji,
+# and default to an empty string if not.
+def check_emoji(emoji):
+    # If the user didn't add an emoji, carry on without it
+    if emoji == None:
+        return clean_text(str(""))
+
+    # Check if the provided character is a valid emoji
+    extracted_emoji = is_emoji_valid(emoji)
+
+    if extracted_emoji is not None:
+        return extracted_emoji
+
 
 class AlertMixin:
-    def error(self, body: "SupportsStr") -> "DeltaGenerator":
+    def error(
+        self,
+        body: "SupportsStr",
+        *,  # keyword-only args:
+        icon: Optional[str] = None,
+    ) -> "DeltaGenerator":
         """Display error message.
 
         Parameters
         ----------
+        icon : None
+            An optional parameter, that adds an emoji to the alert.
+            The default is None.
+            This argument can only be supplied by keyword.
         body : str
             The error text to display.
 
         Example
         -------
-        >>> st.error('This is an error')
+        >>> st.error('This is an error', icon=":rotating_light:")
```

Comment:
Do these types of emoji codes work here? Looking at the validation code, I wouldn't think that they do

Reply:
You're right, they don't. Updated on [a2d0ce4](https://github.com/streamlit/streamlit/pull/4964/commits/a2d0ce4ae5bf29960dd2f4175682405c5785a697)